embedded/sql/sql_grammar.y

```diff
@@ -569,17 +570,32 @@ ds:
         $2.(*SelectStmt).as = $4
         $$ = $2.(DataSource)
     }
+|
+    DATABASES opt_as
+    {
+        $$ = &ListDatabasesStmt{as: $2}
+    }
+|
+    TABLES opt_as
+    {
+        $$ = &ListTablesStmt{as: $2}
+    }
+|
+    tableRef '.' COLUMNS opt_as
+    {
+        $$ = &ListColumnsStmt{tableRef: $1, as: $4}
+    }
+|
+    tableRef '.' INDEXES opt_as
+    {
+        $$ = &ListIndexesStmt{tableRef: $1, as: $4}
+    }
 
 tableRef:
     IDENTIFIER
     {
         $$ = &tableRef{table: $1}
     }
-|
-    IDENTIFIER '.' IDENTIFIER
```

Comment:
This is breaking SQL syntax compatibility - we should confirm if this is allowed.

Alternatively we could just use some simple prefix for schema-based queries,

```sql
SELECT * FROM @table.INDEXES;
SELECT * FROM @DATABASES;
```

that would make the syntax unambiguous.